bikram.md

```diff
@@ -0,0 +1,95 @@
+
+8	Escape Content:
+This means removing any executable code that would cause the browser to do something you don’t want it to. Typically this means removing // &lt;![CDATA[ tags and HTML attributes that cause JavaScript to be evaluated. If you use standard data formats like JSON, you can use standard libraries which have been thoroughly checked by many professionals. However, DO NOT TRY TO DO THIS YOURSELF. Use a known library, or the auto-escaping features of your favorite template library. This needs to be done in the browser and on your server, if you allow users to submit data that is saved into a database.
+
+9	Restrict Testing Environment:
+THUMB Rule. No production data or any form of sensitive data to be used while testing the APIs in the testing environment.
+
+10	Storing Tokens at Client Side:
+There are two ways to save authentication information in the browser:
+	Cookies
+	HTML5 Web Storage
+In each case, you have to trust that browsers are implemented correctly, and that Website A can't somehow access the authentication information for Website B. In that sense, both storage mechanisms are equally secure. Problems can arise in terms of how you use them though.
+•	If you use cookies: The browser will automatically send the authentication information with every request to the API. This can be convenient so long as you know it's happening. Prevention against CSRF is a must while using this technique. It is also strongly recommended to use cookies with HTTPOnly and Secure flags set. This will allow the browser to send along the token for authentication purposes, but won’t expose it to the JavaScript environment. 
+•	If you use HTML5 Web Storage: You have to write JavaScript that manages exactly when and what authentication information is sent.
+
+11	Protect against Cross-Site Request Forgery:
+For resources exposed by RESTful web services, it's important to make sure any PUT, POST, and DELETE request is protected from Cross Site Request Forgery. Typically, one would use a token-based approach. See Cross-Site Request Forgery (CSRF) Prevention Cheat Sheet for more information on how to implement CSRF-protection. CSRF is easily achieved even using random tokens if any XSS exists within your application, so please make sure you understand how to prevent XSS.
+
+12	Insecure direct object references:
+A URL or even a POSTed form should NEVER contain an access control "key" or similar that provides automatic verification. A data contextual check needs to be done, server side, with each request.
+
+13	Enable CORS for all APIs:
+When your API's resources receive requests from a domain other than the API's own domain, you must enable cross-origin resource sharing (CORS) for selected methods on the resource. This amounts to having your API respond to the OPTIONS preflight request with at least the following CORS-required response headers:
```

Comment:
I wonder if this should be the default. Or the default for public APIs (accessible outside of adidas). 

Also isn't this handled by API management @oonovak ?